File: boj/binary_search/로봇 프로젝트.py
// 3649

import sys
input = sys.stdin.readline

while True:
  try:
    hole = int(input()) * 10000000
    n = int(input())
    lego = [int(input()) for _ in range(n)]
    lego.sort()

    left, right = 0, n - 1
    while left < right:
      if lego[left] + lego[right] == hole:
        print('yes %d %d' %(lego[left], lego[right]))
        break
      elif lego[left] + lego[right] > hole: right-=1
      else: left+=1
    
    if left >= right: print('danger')
  except:
    break

File: boj/binary_search/로봇 프로젝트.js
// 3649
// 백준 js 메모리초과 에러

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split("\r\n");

const answer = [];

while (input.length) {
  const hole = +input.shift() * 10000000;
  const len = +input.shift();

  const lego = input.splice(0, len).map(Number).sort((a, b) => a - b);

  let left = 0, right = len - 1;
  while (left < right) {
    if (lego[left] + lego[right] === hole) {
      answer.push(`yes ${lego[left]} ${lego[right]}`);
      break;
    } else if (lego[left] + lego[right] > hole) right--;
    else left++;
  }
  if (left >= right) answer.push('danger');
}

console.log(answer.join('\n'));